PacketDotNet/WakeOnLanPacket.cs

```diff
@@ -129,6 +126,30 @@ public PhysicalAddress DestinationAddress
             }
         }
 
+        /// <summary>
+        /// The Password field is optional, but if present, contains either 4 bytes or 6 bytes.
+        /// If a 4-byte password is present, it will be dissected as an IPv4 address 
+        /// and if a 6-byte password is present, it will be dissected as an Ethernet address.
+        /// </summary>
+        public byte[] Password
+        {
+            get
+            {
+                int passwordLength = HeaderData.Length - WakeOnLanFields.PasswordPosition;
+
+                // If 6-byte length, it will be dissected as an Ethernet address.
+                // If 4-byte length, it will be dissected as an IPv4 address.
+                if (passwordLength == 6 || passwordLength == 4)
+                {
+                    byte[] hwAddress = new byte[passwordLength];
+                    Array.Copy(Header.Bytes, Header.Offset + WakeOnLanFields.PasswordPosition,
+                               hwAddress, 0, hwAddress.Length);
+                    return hwAddress;
+                }
+                return new byte[0];
+            }
```

Comment:
Perhaps a setter can be added here as well? In case someone wants to modify it.

Reply:
Added the setter method as notified.